packages/nx/src/adapter/ngcli-adapter.ts

```diff
@@ -1049,3 +1021,146 @@ function saveProjectsConfigurationsInWrappedSchematic(
+      const { json: packageJson, path: packageJsonPath } =
+        readPluginPackageJson(
+          nodeModule,
+          this.workspaces['resolvePaths'].bind(this.workspaces)()
+        );
+      const executorsFile = packageJson.executors ?? packageJson.builders;
+
+      if (!executorsFile) {
+        throw new Error(
+          `The "${nodeModule}" package does not support Nx executors or Angular Devkit Builders.`
+        );
+      }
+
+      const executorsFilePath = require.resolve(
+        join(dirname(packageJsonPath), executorsFile)
+      );
+      const executorsJson = readJsonFile<ExecutorsJson>(executorsFilePath);
+      const executorConfig: {
+        implementation: string;
+        batchImplementation?: string;
+        schema: string;
+        hasher?: string;
+      } = executorsJson.builders?.[builder];
+      if (!executorConfig) {
+        throw new Error(
+          `Cannot find builder '${builder}' in ${executorsFilePath}.`
+        );
+      }
+      return { executorsFilePath, executorConfig, isNgCompat: true };
+    }
+
+    private readExecutor(
+      nodeModule: string,
+      executor: string
+    ): ExecutorConfig & { isNgCompat: boolean } {
+      try {
+        const { executorsFilePath, executorConfig, isNgCompat } =
+          this.readExecutorsJson(nodeModule, executor);
+        const executorsDir = dirname(executorsFilePath);
+        const schemaPath = this.workspaces['resolveSchema'].bind(
+          this.workspaces
+        )(executorConfig.schema, executorsDir);
+        const schema = normalizeExecutorSchema(readJsonFile(schemaPath));
+
+        const implementationFactory = this.getImplementationFactory<Executor>(
+          executorConfig.implementation,
+          executorsDir
+        );
+
+        const batchImplementationFactory = executorConfig.batchImplementation
+          ? this.getImplementationFactory<TaskGraphExecutor>(
+              executorConfig.batchImplementation,
+              executorsDir
+            )
+          : null;
+
+        const hasherFactory = executorConfig.hasher
+          ? this.getImplementationFactory<CustomHasher>(
+              executorConfig.hasher,
+              executorsDir
+            )
+          : null;
+
+        return {
+          schema,
+          implementationFactory,
+          batchImplementationFactory,
+          hasherFactory,
+          isNgCompat,
+        };
+      } catch (e) {
+        throw new Error(
+          `Unable to resolve ${nodeModule}:${executor}.\n${e.message}`
+        );
+      }
+    }
+
+    private getImplementationFactory<T>(
+      implementation: string,
+      executorsDir: string
+    ): () => T {
+      return this.workspaces['getImplementationFactory'].bind(this.workspaces)(
+        implementation,
+        executorsDir
+      );
+    }
+  }
+
+  return new WrappedWorkspaceNodeModulesArchitectHost(workspace, root);
+}
+
+function getInstalledPackageVersionInfo(
```

Comment:
This function is not needed/used. You can remove it.

Reply:
🤦 my bad!